src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -51,6 +52,11 @@ export async function searchLogGroup(registry: LogStreamRegistry): Promise<void>
 
         const textEditor = await vscode.window.showTextDocument(doc, { preview: false })
         registry.setTextEditor(uri, textEditor)
+        // Initial highlighting of the document and then for any addLogEvent calls.
+        highlightDocument(registry, uri)
+        vscode.workspace.onDidChangeTextDocument((event: vscode.TextDocumentChangeEvent) =>
+            highlightDocument(registry, uri)
+        )
```

Comment:
Let's make sure we only run the highlight logic on the correct document:
```ts
    if (event.document.uri.toString() === doc.uri.toString()) {
         highlightDocument(registry, uri)
    }
```

Reply:
Yup, which is safe but just not efficient. We'll make it more robust later by moving the event listener into `LogStreamDocumentProvider`.